src/main/java/de/tum/in/www1/artemis/service/LectureService.java

```diff
@@ -133,9 +144,35 @@ public SearchResultPageDTO<Lecture> getAllOnPageWithSize(final SearchTermPageabl
      * @param lecture the lecture to be deleted
      */
     public void delete(Lecture lecture) {
+        if (pyrisWebhookService.isPresent()) {
+            Lecture lectureWithAttachmentUnits = lectureRepository.findByIdWithLectureUnitsAndAttachmentsElseThrow(lecture.getId());
+            List<AttachmentUnit> attachmentUnitList = lectureWithAttachmentUnits.getLectureUnits().stream()
+                    .filter(lectureUnit -> lectureUnit instanceof AttachmentUnit && ((AttachmentUnit) lectureUnit).getAttachment().getLink().endsWith(".pdf"))
+                    .map(lectureUnit -> (AttachmentUnit) lectureUnit).toList();
+            if (!attachmentUnitList.isEmpty()) {
+                pyrisWebhookService.get().deleteLectureFromPyrisDB(attachmentUnitList);
+            }
+        }
         Channel lectureChannel = channelRepository.findChannelByLectureId(lecture.getId());
         channelService.deleteChannel(lectureChannel);
         lectureRepository.deleteById(lecture.getId());
     }
 
+    /**
+     * Ingest the lectures when triggered by the ingest lectures button
+     *
+     * @param lectures set of lectures to be ingested
+     * @return returns the job token if the operation is successful else it returns null
+     */
+    public boolean ingestLecturesInPyris(Set<Lecture> lectures) {
+        if (pyrisWebhookService.isPresent()) {
+            List<AttachmentUnit> attachmentUnitList = lectures.stream().flatMap(lec -> lec.getLectureUnits().stream()).filter(unit -> unit instanceof AttachmentUnit)
```

Comment:
For deletion you only handle attachments that end with `.pdf` but you do not filter here. Why does this inconsistency exist?

Reply:
The filtering happens at the level of deletion and the addition, because they are called from different places and like that the filtering is consistent.